pkg/scalers/cassandra_scaler.go

```diff
@@ -157,9 +169,79 @@ func parseCassandraMetadata(config *scalersconfig.ScalerConfig) (*CassandraMetad
 		return nil, fmt.Errorf("no password given")
 	}
 
+	if err = parseCassandraTLS(config, meta); err != nil {
+		return meta, err
+	}
+
 	meta.triggerIndex = config.TriggerIndex
 
-	return &meta, nil
+	return meta, nil
+}
+
+func createTempFile(prefix string, content string) (string, error) {
+	f, err := os.CreateTemp("/tmp/cassandra/", prefix+"-*.pem")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(content)
+	if err != nil {
+		return "", err
+	}
+
+	return f.Name(), nil
+}
+
+func parseCassandraTLS(config *scalersconfig.ScalerConfig, meta *CassandraMetadata) error {
+	meta.enableTLS = false
+	if val, ok := config.AuthParams["tls"]; ok {
+		val = strings.TrimSpace(val)
+		if val == tlsEnable {
+			certGiven := config.AuthParams["cert"] != ""
+			keyGiven := config.AuthParams["key"] != ""
+			caCertGiven := config.AuthParams["ca"] != ""
+			if certGiven && !keyGiven {
+				return errors.New("no key given")
+			}
+			if keyGiven && !certGiven {
+				return errors.New("no cert given")
+			}
+			if !keyGiven && !certGiven {
+				return errors.New("no cert/key given")
+			}
+
+			certFilePath, err := createTempFile("cert", config.AuthParams["cert"])
```

Comment:
we should use random identifiers to avoid collisions when there are more than 1 scaler using this approach. You can see how it's done in kafka for kerberos: https://github.com/kedacore/keda/blob/main/pkg/scalers/kafka_scaler.go#L430-L454

Reply:
createTempFile will create a file with cert-<uuid>, so we should be Good.

Reply:
yeah, but I meant about the name prefix. Basically if I have 2 cassandra scalers with different certs, they will be in conflict because the cert paths are the same. When I said following that approach was for using random names for each file to prevent colissions